### DISCUSSION TITLE
Add the ability to change offset during gameplay

### DISCUSSION THREAD

**Discussion by Bubbleman532** [ppy/osu#25000] [Ideas] [answered]
Currently there's no way to quickly adjust offset mid-map as there is in stable. Being able to adjust offset using the +/- keys by 5/1ms during breaks was an incredibly useful feature in multiplayer if you were sightreading a map, and it being missing means players are forced to play through with a sub-optimal offset. It's also useful to have if a map intro is slow, so would lower time spent finding a map offset tremendously by not having to sit through the intro over and over.

As an aside, the pre-map offset slider should also have a way to be adjusted in +/-1ms increments, tho I can open up another feature request for this if the separation is needed.

**Comment by bdach** [comment]
> Currently there's no way to quickly adjust offset mid-map as there is in stable

Just as an open-ended question from a village idiot: is this not a gameplay integrity concern? Can being able to mess with offset during gameplay be an unfair benefit? I'd imagine it could be, in contexts other than sightreading a map for the first time ever.

> As an aside, the pre-map offset slider should also have a way to be adjusted in +/-1ms increments, tho I can open up another feature request for this if the separation is needed.

That would be best I think.

**Reply by Bubbleman532** [reply]
offset changing doesn't inherently change gameplay so no, I don't think there could be anything to be gained, and soft proof of that is that it already exists and nobody has done anything with it. Most notable maybe is on album marathons like The Unforgiving where each song is timed slightly off but there's a break between each where you have the chance to change offset

**Reply by peppy** [reply]
i'm pretty sure stable doesn't allow changes after the first x seconds

**Reply by Bubbleman532** [reply]
This is fine, but offset in stable can be changed during break times and this ability is missing in lazer

**Reply by peppy** [reply, marked as answer]
Tracking at https://github.com/ppy/osu/issues/25007
